src/resolvers/projectResolver.ts

```diff
@@ -567,221 +573,88 @@ export class ProjectResolver {
-        //   [key: string]: 'NULLS FIRST' | 'NULLS LAST';
-        // } = {
-        //   ASC: 'NULLS LAST',
-        //   DESC: 'NULLS FIRST',
-        // };
-        //
-        // query.orderBy(
-        //   `project.${orderBy.field}`,
-        //   orderBy.direction,
-        //   acceptGivDirection[orderBy.direction],
-        // );
-        query.andWhere(
-          `project.${orderBy.field} IS${
-            orderBy.direction === OrderDirection.DESC ? '' : ' NOT'
-          } NULL`,
-        );
-        query.orderBy(
-          `project.${OrderField.CreationDate}`,
-          OrderDirection.DESC,
-        );
-        break;
-      case OrderField.Verified:
-        query.andWhere(`project.${orderBy.field} = true`);
-        query.orderBy(`project.${OrderField.CreationDate}`, orderBy.direction);
-        break;
-      case OrderField.GIVPower:
-        query
-          .orderBy('projectPower.totalPower', orderBy.direction, 'NULLS LAST')
-          .addOrderBy(
-            `project.${OrderField.CreationDate}`,
-            OrderDirection.DESC,
-          );
-        break;
-      default:
-        query.orderBy(`project.${orderBy.field}`, orderBy.direction);
-        break;
-    }
-
-    const [projects, totalCount] = await query
-      .take(take)
-      .skip(skip)
-      .getManyAndCount();
-    return { projects, totalCount, categories };
-  }
 
-  @Query(returns => AllProjects)
-  async allProjects(
-    @Args()
-    {
+    const projectsCountQuery = filterProjectsQuery(
       limit,
       skip,
       searchTerm,
       category,
       mainCategory,
       filters,
       sortingBy,
-      admin,
       connectedWalletUserId,
-    }: GetProjectsArgs,
-    @Ctx() { req: { user } }: MyContext,
-  ): Promise<AllProjects> {
-    const categories = await Category.find();
-    let query = this.projectRepository
-      .createQueryBuilder('project')
-      .leftJoinAndSelect('project.status', 'status')
-      .leftJoinAndSelect('project.users', 'users')
-      .leftJoinAndSelect('project.addresses', 'addresses')
-      .leftJoinAndSelect('project.organization', 'organization')
-      // you can alias it as user but it still is mapped as adminUser
-      // like defined in our project entity
-      .innerJoin('project.adminUser', 'user')
-      .addSelect(publicSelectionFields) // aliased selection
-      .leftJoinAndSelect(
-        'project.categories',
-        'categories',
-        'categories.isActive = :isActive',
-        { isActive: true },
-      )
-      .leftJoinAndSelect('categories.mainCategory', 'mainCategory')
-      .leftJoin('project.projectPower', 'projectPower')
-      .addSelect([
-        'projectPower.totalPower',
-        'projectPower.powerRank',
-        'projectPower.round',
-      ])
-      .where(
-        `project.statusId = ${ProjStatus.active} AND project.listed = true`,
-      );
+      user,
+    );
+    const filtersResultHashTask = projectsFiltersThreadPool.queue(hasher =>
```

Comment:
Why we run two queries, one for count and one for fetching the projects?

Reply:
I will test GetManyAndCount() query with 1 cache and let you know. I think I avoided it because I was getting errors with the cache. Will report back on this comment.